Updates on PMA\Template class

gzzhanghao opened the pull request:
Updates on Template class.
- Add `trim` method to Template class. Now the usage is changed to: 
  `Template::get($template_script_name)->render($data, [$trim = true]);`
  The `$trim` parameter in the render function indicate whether it should eliminate white spaces between HTML tags and its values. This feature is set to ON by default; You can disable it by simply passing `false`.
- `trim` is also provided as a static method.
- Update tests for the Template class.
- Add templates for db_designer.lib.php.
- Change extensions of templates to ".pthml".

Tithugues commented:
Hi,
Did you create this for a specific need?

Tithugues commented on libraries/Template.class.php:
```diff
@@ -0,0 +1,44 @@
+                include $template;
+            } else {
+                throw new \LogicException(
+                    'The template "' . $template . '" not found.'
```
I think that we should use translation here.

zixtor replied:
I think it is intended more for the developers if they do happen to type a wrong template name, so we may choose to skip translating it. Its thrown again in catch, so if a user does see a Fatal error, can just copy it and report to us, as user not expected to really do something about it.

zixtor commented:
@Tithugues It must be related to https://wiki.phpmyadmin.net/pma/GSoC_2015_Projects#Codebase_Improvements:_OOP_and_refactoring_-_Zhang

zixtor commented:
@gzzhanghao Please update it such as to have only changes on top of the current master. (so it will just include addition of trim function and all template class unit tests)

gzzhanghao committed:
Improve template system

- Introduce `trim` method for templates
- Complete template of the designer module
- Update tests for template system
- Use phtml as template files extension

Signed-off-by: Jason <jason.daurus@gmail.com>

zixtor commented:
@gzzhanghao It would have been better had you done separate things in a new PR, but for now, please update the title/description as well for this PR, maybe as mentioned in your commit message.

zixtor commented on templates/designer/database_tables.phtml:
```diff
@@ -0,0 +1,140 @@
+                                if (strstr($tab_column[$t_n]["TYPE"][$j], 'char')
+                                    || strstr($tab_column[$t_n]["TYPE"][$j], 'text')
+                                ) {
+                                    $type = '_char';
```
This must be .= here and at other similar places.

zixtor commented:
Overall looks very good to me, make the changes as mentioned in above comments, and its ready to merge.

gzzhanghao committed:
Fix database_tables.phtml

Signed-off-by: Jason <jason.daurus@gmail.com>

zixtor commented on templates/designer/database_tables.phtml:
```diff
@@ -0,0 +1,140 @@
+                            <?php if (isset($tables_pk_or_unique_keys[$t_n . "." . $tab_column[$t_n]["COLUMN_NAME"][$j]])) : ?>
+                                <img src="<?php echo $_SESSION['PMA_Theme']->getImgPath('pmd/FieldKey_small.png'); ?>" alt="*" />
+                            <?php else :
+                                $type = 'pma/Field_small';
```
Typo here, pma -> pmd..

gzzhanghao committed:
Fix an typo cause FieldKey_*.png not resolve correctly

Signed-off-by: Jason <jason.daurus@gmail.com>

gzzhanghao commented:
@zixtor The typo problem was fixed in the newest commit. Please merge :-)

zixtor commented on scripts/update-po:
```diff
@@ -18,7 +18,7 @@ xgettext \
     --keyword=__ --keyword=_pgettext:1c,2 --keyword=_ngettext:1,2 \
     --copyright-holder="phpMyAdmin devel team" \
-    `find . -name '*.php' -not -path './test/*' -not -path './po/*' -not -path './release/*' | sort`
+    `find . -name '*.php' -o -name '*.phtml' -not -path './test/*' -not -path './po/*' -not -path './release/*' | sort`
```
@gzzhanghao One more thing, this find now seems to be resulting in ./test/* files as well, using regex seems ok to me:-
find -regex '.*\.\(php\|phtml\)' -not -path './test/*' -not -path './po/*' -not -path './release/*' | sort | wc -l

gzzhanghao committed:
Fix update-po script

Signed-off-by: Jason <jason.daurus@gmail.com>

gzzhanghao commented:
@zixtor Thank you for providing a idea on solving this problem. Now the problem should have been fixed :-)

zixtor commented:
Thanks!
